scripts/publish.sh

```diff
@@ -44,45 +44,87 @@ function release {
   git push --follow-tags
   npx conventional-github-releaser -p angular
+
+  echo "Beginning release $PKG_NAME@$VERSION"
   npm publish --access public
+
+  echo "Release dist for $PKG_NAME@$VERSION"
+  ls dist
+
   # `npm view name@x.x.x version` will return an empty string if not available
   until [ "$(npm view $PKG_NAME@$VERSION version)" != "" ];
   do
     echo "Waiting for publish to complete"
     sleep 3
   done
+
   # update all workspaces from the workspace root (../..) with the new version
-  # make sure publish.sh is called in topological order, `lerna run` does this
-  npx lerna-update-wizard --non-interactive --lazy --dependency $PKG_NAME@$VERSION ../..
+  # make sure publish.sh is called in topological order, `lerna ls --toposort` does this
+  DEPENDANT_PKGS=$(npx lerna ls --graph --toposort --scope @mux/* |
+    jq -r "to_entries[] | select(.value[] | contains(\"$PKG_NAME\")) | .key")
+  scope=""
+  for name in ${DEPENDANT_PKGS}; do
+    scope+="--scope $name "
+  done
+  npx lerna exec $scope -- npm pkg set dependencies.$PKG_NAME=$VERSION
+  yarn --cwd="../.." install
+  echo "Ending release $PKG_NAME@$VERSION"
 };
 
 function canary {
   PKG_NAME=$(cat package.json | jq -r '.name')
   PKG_VERSION=$(cat package.json | jq -r '.version')
+
   # get last published version from NPM without alpha / beta, remove -SHA hash
+
+  # debug jq command at https://jqplay.org/s/Wwjv5hVUCL0
+  #
+  #   1. remove alpha or beta versions
+  #   2 & 3. convert to { version: 'x.x.x', dist: 'canary', build: x }
+  #   4. sort first by `version` then by `build` number
+  #   5. put back together to a string `x.x.x-canary.x`
+  #   6. pick the last item in the array
+
   LAST_VERSION=$(npm view $PKG_NAME versions --json |
-    jq -r '. - map(select(contains("alpha") or contains("beta"))) | last' |
-    sed -r 's/-[a-z0-9]{7}$//g')
-  # GH actions made this a `null` string, replace it with a null value
+    jq -r '. - map(select(contains("alpha") or contains("beta")))
+      | map(capture("(?<version>\\d+\\.\\d+\\.\\d+)(-(?<dist>[a-z]+))?(\\.(?<build>\\d+))?"))
+      | map(.build? |= (. // 0 | tonumber))
+      | sort_by(.version, .build)
+      | map(.version + "-" + (.dist // "latest") + "." + (.build|tostring))
+      | last')
+
+  # if json is empty GH actions made this a `null` string, replace it with a null value
   LAST_VERSION=$(echo $LAST_VERSION | sed "s/null//g")
+
   # default to local package version if no last version was found on NPM
   PRE_VERSION=$(npx semver ${LAST_VERSION:-$PKG_VERSION} -i prerelease --preid canary)
   VERSION=$PRE_VERSION-$(git rev-parse --short HEAD)
-  echo "Beginning canary release for $PKG_NAME@$VERSION"
+
+  echo "Beginning release $PKG_NAME@$VERSION"
   npm --no-git-tag-version version $VERSION
   npm publish --tag canary --access public
-  echo "canary release dist used for $PKG_NAME@$VERSION"
+
+  echo "Release dist for $PKG_NAME@$VERSION"
```

Comment:
can we make it clear that this is a canary release here?

Reply:
it's in the version itself ;) 
Beginning release @mux/playback-core@0.9.1-canary.11-f46ae2d